### PR TITLE
feat: Add more notebook samples for documentation

### PR DESCRIPTION
**Pull request by serena-ruan**

### CODE REVIEW

File: notebooks/samples/CognitiveServices - Celebrity Quote Analysis.ipynb [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "\n",
     "bingSearch = BingImageSearch()\\\n",
     "  .setSubscriptionKey(BING_IMAGE_SEARCH_KEY)\\\n",
+    "  .setUrl(\"https://api.bing.microsoft.com/v7.0/images/search\")\\\n",
```
Comment (mhamilton723) [inline notebooks/samples/CognitiveServices - Celebrity Quote Analysis.ipynb:62]: You can remove this now

**Review by imatiach-msft — DISMISSED** [review]

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
Sent comments in teams

**Comment by eisber** [comment]
For VW:

Advantages: you can also mention it supports distributed learning (the parallelization is controlled through the number of data partitions)
Quantile regression: I'd add a link to wikipedia or some nice blog article. One can do really interesting things w/ quantile regression (e.g. predict the 75% quantile of transportation cost)
maybe pull the boston sample up?

File: notebooks/samples/CognitiveServices - Celebrity Quote Analysis.ipynb (hunk 1 above)
Comment (serena-ruan) [reply notebooks/samples/CognitiveServices - Celebrity Quote Analysis.ipynb]: Right! Done.

**Comment by serena-ruan** [comment]
> For VW:
> 
> Advantages: you can also mention it supports distributed learning (the parallelization is controlled through the number of data partitions)
> Quantile regression: I'd add a link to wikipedia or some nice blog article. One can do really interesting things w/ quantile regression (e.g. predict the 75% quantile of transportation cost)
> maybe pull the boston sample up?

Done! Thanks for the comment!

File: notebooks/samples/Cognitive Services Overview.ipynb [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+    "\n",
+    "### Decision\n",
+    "[**Anomaly Detector**](https://azure.microsoft.com/en-us/services/cognitive-services/anomaly-detector/)\n",
+    "- Anomaly status of latest point: generates a model using preceding points and determines whether the latest point is anomalous ([Scala](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/scala/com/microsoft/ml/spark/cognitive/DetectLastAnomaly.html), [Python](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/pyspark/mmlspark.cognitive.html#module-mmlspark.cognitive.DetectLastAnomaly))\n",
+    "- Find anomalies: generates a model using an entire series and finds anomalies in the series ([Scala](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/scala/com/microsoft/ml/spark/cognitive/DetectAnomalies.html), [Python](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/pyspark/mmlspark.cognitive.html#module-mmlspark.cognitive.DetectAnomalies))\n",
+    "\n",
+    "### Web Search\n",
```
Comment (mhamilton723) [inline notebooks/samples/Cognitive Services Overview.ipynb:63]: Web Search -> Search

File: notebooks/samples/Cognitive Services Overview.ipynb [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+    "- [Bing Image search](https://azure.microsoft.com/en-us/services/cognitive-services/bing-image-search-api/) ([Scala](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/scala/com/microsoft/ml/spark/cognitive/BingImageSearch.html), [Python](https://mmlspark.blob.core.windows.net/docs/1.0.0-rc3/pyspark/mmlspark.cognitive.html#module-mmlspark.cognitive.BingImageSearch))\n",
+    "- [Azure Cognitive search](https://docs.microsoft.com/en-us/azure/search/search-what-is-azure-search)\n"
```
Comment (mhamilton723) [inline notebooks/samples/Cognitive Services Overview.ipynb:65]: Could you add corresponding scala snd python docs links?

File: notebooks/samples/LightGBM Overview.ipynb [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+    "test_data = featurizer.transform(test)['Bankrupt?', 'features']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.show(10)"
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:153]: Can remove

File: notebooks/samples/LightGBM Overview.ipynb [hunk 5]
```diff
@@ -0,0 +1,548 @@
...
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.groupBy(\"Bankrupt?\").count().show()"
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:169]: show -> display

File: notebooks/samples/LightGBM Overview.ipynb [hunk 6]
```diff
@@ -0,0 +1,548 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mmlspark.lightgbm import LightGBMClassificationModel\n",
+    "model.saveNativeModel(\"/lgbmcmodel\")\n",
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:205]: perhaps we can call this lgbmclassifier.model and add a cried markdown description that this allows you to extract the underlying lightGBM model for fast deployment after you train on spark

File: notebooks/samples/LightGBM Overview.ipynb [hunk 7]
```diff
@@ -0,0 +1,548 @@
...
+    "fi = pd.Series(feature_importances,index = feature_cols)\n",
+    "fi = fi.sort_values(ascending = True)\n",
+    "f_index = fi.index\n",
+    "f_values = fi.values\n",
+    " \n",
+    "# print feature importances \n",
+    "print ('f_index:',f_index)\n",
+    "print ('f_values:',f_values)\n",
+    "\n",
+    "# plo\n",
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:235]: typo: plo -> plot

File: notebooks/samples/LightGBM Overview.ipynb [hunk 8]
```diff
@@ -0,0 +1,548 @@
...
+    "label_col = 'labels'\n",
+    "# read in CSV file\n",
+    "df2 = spark.read.format('csv').option('inferSchema', True) \\\n",
+    "    .load(\"wasbs://publicwasb@mmlspark.blob.core.windows.net/lightGBMRanker_rank_train_query.csv\") \\\n",
+    "    .withColumn('index', monotonically_increasing_id()) \\\n",
+    "    .withColumn(query_col, explode(create_rows_udf('_c0', 'index'))) \\\n",
+    "    .withColumn('iid', monotonically_increasing_id()) \\\n",
+    "    .drop('_c0', 'index') \\\n",
+    "    .join(df1, 'iid').drop('iid') \\\n",
+    "    .withColumnRenamed('label', label_col)"
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:457]: Can we save the result of this to parquet in the public blob then read it in so the demo is smaller?

File: notebooks/samples/LightGBM Overview.ipynb [hunk 9]
```diff
@@ -0,0 +1,548 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dt1 = spark.read.format('libsvm') \\\n",
+    "    .load(\"wasbs://publicwasb@mmlspark.blob.core.windows.net/lightGBMRanker_rank_test.libsvm\") \\\n",
+    "    .withColumn('iid', monotonically_increasing_id())\n",
+    "dt2 = spark.read.format('csv').option('inferSchema', True) \\\n",
```
Comment (mhamilton723) [inline notebooks/samples/LightGBM Overview.ipynb:527]: likewise here

File: notebooks/samples/Vowpal Wabbit Overview.ipynb [hunk 10]
```diff
@@ -0,0 +1,659 @@
+{
```
Comment (mhamilton723) [inline notebooks/samples/Vowpal Wabbit Overview.ipynb:1]: Nit to keep with the style of others lets make title Vowpal Wabbit - Overview. Likewise for other NBs

File: notebooks/samples/Vowpal Wabbit Overview.ipynb [hunk 11]
```diff
@@ -0,0 +1,659 @@
...
+    "test_data = featurizer.transform(test)[\"target\", \"features\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.groupBy(\"target\").count().show()"
```
Comment (mhamilton723) [inline notebooks/samples/Vowpal Wabbit Overview.ipynb:151]: display

File: notebooks/samples/Vowpal Wabbit Overview.ipynb [hunk 12]
```diff
@@ -0,0 +1,659 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = spark.read.parquet(\"wasbs://publicwasb@mmlspark.blob.core.windows.net/AdultCensusIncome.parquet\")\n",
+    "data = data.select([\"education\", \"marital-status\", \"hours-per-week\", \"income\"])\n",
+    "train, test = data.randomSplit([0.75, 0.25], seed=123)\n",
+    "display(train.limit(10))"
```
Comment (mhamilton723) [inline notebooks/samples/Vowpal Wabbit Overview.ipynb:224]: no need for limit

File: notebooks/samples/Vowpal Wabbit Overview.ipynb [hunk 13]
```diff
@@ -0,0 +1,659 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Making predictions\n",
+    "test = test.withColumn(\"label\", when(col(\"income\").contains(\"<\"), 0.0).otherwise(1.0))\n",
+    "prediction = vw_trained.transform(test)\n",
+    "display(prediction.limit(10))"
```
Comment (mhamilton723) [inline notebooks/samples/Vowpal Wabbit Overview.ipynb:308]: no need for limit

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
Great Job! Mostly just little things left.

Two larger questions:

 there are a lot of cache count and repartitions going on in VW code. Would you be able to try removing some of these to see if they are necessary? These operations in general will slow things down. 
 
 I will also send over Jack Gerrits example on Vowpal Wabbit Contextual Bandit code when available, (we don't have to block on this though it can be a separate PR)

**Pull request merged**